src/pages/workspaces/workspace/analysis/tool-utils.ts

```diff
@@ -0,0 +1,101 @@
+import _ from 'lodash/fp'
+import { isCromwellAppVisible } from 'src/libs/config'
+import * as Utils from 'src/libs/utils'
+import { getExtension } from 'src/pages/workspaces/workspace/analysis/file-utils'
+
+
+export type ToolLabel = 'Jupyter' | 'RStudio' | 'spark' | 'Azure' | 'Galaxy' | 'terminal' | 'Cromwell'
+export const toolLabelTypes: Record<ToolLabel, ToolLabel> = {
+  Jupyter: 'Jupyter',
+  RStudio: 'RStudio',
+  terminal: 'terminal',
+  spark: 'spark',
+  Azure: 'Azure',
+  Galaxy: 'Galaxy',
+  Cromwell: 'Cromwell'
+}
+// The label here matches the leonardo `tool` label for runtimes
+export interface Tool {
+  label: ToolLabel,
+  ext?: string[],
+  imageIds?: string[],
+  defaultImageId?: string,
+  defaultExt?: string,
+  isNotebook?: boolean
+  appType?: string,
+  isLaunchUnsupported?: boolean
+  isPauseUnsupported?: boolean
+  isAzureCompatible?: boolean,
+  isAppHidden?: boolean
```

Comment:
I wonder if the fact that we need all these optional fields may be code smell? We initially defined `tools` as nested objects because JS didn't have types, enums, etc. Now that we have those, I'd suggest considering taking advantage of them. Perhaps the `Tool` interface can define only the common fields and then there can be extensions of it like `RStudioTool` and `JupyterTool`, which defines their specific properties as needed. I think TypeScript allows the equivalent of pattern matching in Scala which could perhaps then be leveraged to require those tool-specific fields like you did with `CloudProvider` typing [somewhere else](https://github.com/DataBiosphere/terra-ui/pull/3447/files#diff-79707a07ad14b98951a5cb25209dbd412a61b4f23f71c2dc29f64ce961908958R6).

Reply:
Yeah, this looks like it could benefit from the same treatment as the Workspace type. Some BaseTool extended by JupyterTool, RStudioTool, etc. and a Tool type that is the union of all of those.

Reply:
Hmm... I would agree with this slightly.  There are a few complicatting (below) factors though that lead to me wanting to leave this as a TODO for now though. It may be worth discussing this one.
1. Over half of the fields are optional boolean flags whose exclusion from the data (aka undefined) is interpreted as falsey, instead of explicitly including the flag in every tool entry.
2. I don't like over-eagerly making this optimization here since it would cascade to a lot of tool-related code that is not yet tested, since consuming the fields will require changing product code via predicates (aka isJupyterTool). 
3. The analysisDuplicator does not heavily rely on tool code

There will be future tickets (such as testing `Analyses`, or the `CloudEnvironmentModal`, aka context bar details view) that rely on logic such as `if this tool x else y` that will benefit from this in a testable way.